vendor/github.com/vektah/gqlparser/v2/ast/decode.go

```diff
@@ -0,0 +1,198 @@
+package ast
+
+import (
+	"encoding/json"
+)
+
+func UnmarshalSelectionSet(b []byte) (SelectionSet, error) {
+	var tmp []json.RawMessage
+
+	if err := json.Unmarshal(b, &tmp); err != nil {
+		return nil, err
+	}
+
+	var result = make([]Selection, 0)
```

Comment:
[nit] I guess we know the size?
```suggestion
	var result = make([]Selection, 0, len(tmp))
```